src/main/webapp/components/Table.js

```diff
@@ -145,6 +146,21 @@ export const Table = hh(class Table extends Component {
     return btn;
   };
 
+  roleSelection = (cell,row) => {
+    return this.props.isViewer ? null :
+      div({className : "roles-container"}, [
+        span({}, [formatRoleName(row.roles)]),
+        button({
+            className: "btn btn-default btn-sm edit",
+            onClick: this.props.editRole(row),
+            disabled: !this.props.isAdmin,
+            title: "Edit"
+          }, [
+            span({className: "glyphicon glyphicon-pencil"}, [])
+          ]
+      )]);
+  };
+
```

Comment:
So, technically, this should really not be part of the table code. Table should be very generic - by adding lots of other-feature-specific-functionality to it, we're breaking encapsulation. What this should be is a function defined in the RoleManagement component that is passed down into Table so that table knows what to do with it, i.e., letting table say, use function X for `dataFormat` in the case of column `X`. I'm afraid we might be a little too far down this rabbit hole right now ... table is super content dependant and getting worse.

Reply:
One thought, perhaps for a tech-debt story, is to refactor Table so it doesn't have to deal with all of the headers on its own. Maybe if we passed in a list of header values (each one either as a value or a function), then table could iterate through them and set the props for each one.

Reply:
That makes sense. This component started to grow and gain functionality uncontrolled. Also if its ok, it would be really cool to use React Bootstrap Table 2.

Reply:
No objections from me on updating the react bootstrap table. We're on `"react-bootstrap-table": "4.3.1"` now, not sure what it would take to migrate to `react-bootstrap-table2`. I would suggest that be a very separate task, hopefully for obvious reasons.

Reply:
I've created a story for this task [BTRX-716](https://broadinstitute.atlassian.net/browse/BTRX-716)